en/1/04-structs.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Structs
+actions: ['checkAnswer', 'hints']
+material:
+  editor:
+    language: rust
+    startingCode: |
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+
+      // start here
+
+      #[multiversx_sc::contract]
+      pub trait ZombiesContract {
+
+        #[init]
+        fn init(&self) {
+          self.dna_digits().set(16u8);
+        }
+
+        #[storage_mapper("dna_digits")]
+        fn dna_digits(&self) -> SingleValueMapper<u8>;
+      }
+    answer: >
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+
+      #[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, TypeAbi)]
+      pub struct Zombie<M: ManagedTypeApi> {
+          name: ManagedBuffer<M>,
+          dna: u64,
+      }
+
+      #[multiversx_sc::contract]
+      pub trait ZombiesContract {
+
+        #[init]
+        fn init(&self) {
+          self.dna_digits().set(16u8);
+        }
+
+        #[storage_mapper("dna_digits")]
+        fn dna_digits(&self) -> SingleValueMapper<u8>;
+      }
+---
+
+# The dynamic allocation problem
```

Comment:
Why not rather calling this section "Managed types" as this seems to be the concept we introduce?

Also, I think we should explain why we call them "managed" types. If I get it right, this is because the node managed them instead of Rust directly.